### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by PetrAPConsulting**
I added description and script for batch of supported files processing. It's how I use it and it could be useful for others.

### CODE REVIEW

**Comment by PetrAPConsulting** [comment]
> @PetrAPConsulting please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree

**Comment by afourney** [comment]
I love this idea, but I also want to keep the README short and simple. 

@gagb What do you think about either hosting samples, having a longer document (not sure what to call it), or even hosting documentation?

As this repo continues to grow, I'd like to maintain the simplicity at the front door -- first-run experience.

**Comment by gagb** [comment]
> I love this idea, but I also want to keep the README short and simple.
> 
> @gagb What do you think about either hosting samples, having a longer document (not sure what to call it), or even hosting documentation?
> 
> As this repo continues to grow, I'd like to maintain the simplicity at the front door -- first-run experience.

Several options:
- use collapsed section for this example: https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-collapsed-sections
- create and host docs -- I'll create an issue for this

**Comment by PetrAPConsulting** [comment]
Hi,

I completely understand the goal of keeping the README as simple as possible. As a suggested action, I incorporated the changes and collapsed the proposed new paragraph. The text is definitely clearer and easier to navigate.

Let me know if it meets requirements.

File: README.md [hunk 1]
```diff
@@ -60,7 +60,55 @@ print(result.text_content)
...
+        
+        print(f"Successfully converted {file} to {md_file}")
+    except Exception as e:
+        print(f"Error converting {file}: {str(e)}")
+
+print("\nAll conversions completed!")
+```
+2. Place the script in the same directory as your files
+3. Install required packages: like openai
+4. Run script ```bash python3 convert.py ```
```
Comment (gagb) [inline README.md:103]: `python convert.py`?

File: README.md [hunk 2]
```diff
@@ -60,7 +60,55 @@ print(result.text_content)
 docker build -t markitdown:latest .
 docker run --rm -i markitdown:latest < ~/your-file.pdf > output.md
 ```
+<details>
+    
+<summary>Batch Processing Multiple Files</summary>
 
+This extension allows you to convert multiple files to markdown format in a single run. The script processes all supported files in a directory and creates corresponding markdown files.
```
Comment (gagb) [inline README.md:67]:
```suggestion
This example shows how to convert multiple files to markdown format in a single run. The script processes all supported files in a directory and creates corresponding markdown files.
```

File: README.md [hunk 3]
```diff
@@ -60,7 +60,55 @@ print(result.text_content)
...
+
+### Features
+
+- Converts multiple files in one operation
+- Supports various file formats (.pptx, .docx, .pdf, .jpg, .jpeg, .png etc. you can change it)
+- Maintains original filenames (changes extension to .md)
+- Includes GPT-4o-latest image descriptions when available
+- Continues processing if individual files fail
```
Comment (gagb) [inline README.md:75]:
```suggestion
```

File: README.md [hunk 4]
```diff
@@ -60,7 +60,55 @@ print(result.text_content)
...
+
+### Usage
+1. Create a Python script (e.g., convert.py):
```
Comment (gagb) [inline README.md:78]:
```suggestion
```

File: README.md [hunk 5]
```diff
@@ -60,7 +60,55 @@ print(result.text_content)
...
+print("\nAll conversions completed!")
+```
+2. Place the script in the same directory as your files
+3. Install required packages: like openai
+4. Run script ```bash python3 convert.py ```
+
+- The script processes all supported files in the current directory
+- Original files remain unchanged
+- New markdown files are created with the same base name
+- Progress and any errors are displayed during conversion
```
Comment (gagb) [inline README.md:108]:
```suggestion
Note that original files will remain unchanged and new markdown files are created with the same base name.
```

File: README.md (hunk 1 above)
Comment (PetrAPConsulting) [reply README.md:96]: sorry, I run it on Mac therefore python3 but of course on Windows it's different

File: README.md [hunk 6]
```diff
@@ -60,7 +60,55 @@ print(result.text_content)
...
+- Converts multiple files in one operation
+- Supports various file formats (.pptx, .docx, .pdf, .jpg, .jpeg, .png etc. you can change it)
+- Maintains original filenames (changes extension to .md)
+- Includes GPT-4o-latest image descriptions when available
+- Continues processing if individual files fail
+
+### Usage
+1. Create a Python script (e.g., convert.py):
+```python
+from markitdown import MarkItDown
```
Comment (gagb) [inline README.md:71]:
```suggestion
```python convert.py
from markitdown import MarkItDown
```

**Comment by gagb** [comment]
@PetrAPConsulting thanks, please see my feedback to simplify further.

File: README.md [hunk 7]
```diff
@@ -63,7 +63,43 @@ print(result.text_content)
...
 
+This example shows how to convert multiple files to markdown format in a single run. The script processes all supported files in a directory and creates corresponding markdown files.
+
+
+```python convert.py
+from markitdown import MarkItDown
+from openai import OpenAI
+import os
+client = OpenAI(api_key="your-api-key-here")
+md = MarkItDown(mlm_client=client, mlm_model="gpt-4o-2024-11-20")
```
Comment (gagb) [inline README.md:78]:
```suggestion
md = MarkItDown(llm_client=client, llm_model="gpt-4o-2024-11-20")
```

Comment (PetrAPConsulting) [reply README.md]: Hello @gagb, this will not work. mlm parametr is define directly in the library's code. It can't be changed.

Comment (gagb) [reply README.md]: It will also make your example consistent with the Python API above in the readme

Comment (PetrAPConsulting) [reply README.md]: sorry, I was not aware you are going to change this parameters' terminology in the code

**Pull request merged**